### PR TITLE
LOG-1659: Update dev images to use public bases

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* Replaces ocp specific base images with those that are publicly available with RH or other subscriptions

### Links
* https://issues.redhat.com/browse/LOG-1659

cc @alanconway

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ### This is a generated file from Dockerfile.in ###
 #@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:rhel_8_golang_1.16)
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.8 AS builder
+FROM registry.access.redhat.com/ubi8/go-toolset AS builder
```
Comment (vimalk78) [inline Dockerfile:3]: suggest to use
```
registry.redhat.io/ubi8/go-toolset:1.16.12 
```

File: Dockerfile [hunk 2]
```diff
@@ -29,14 +29,15 @@ COPY ${REMOTE_SOURCE}/apis ./apis
 COPY ${REMOTE_SOURCE}/controllers ./controllers
 COPY ${REMOTE_SOURCE}/internal ./internal
 
+USER 0
```
Comment (vimalk78) [inline Dockerfile:32]: why we need privileged access in CLO?

File: Dockerfile [hunk 3]
```diff
@@ -29,14 +29,15 @@ COPY ${REMOTE_SOURCE}/apis ./apis
...
 RUN make build
 
 
 #@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-cli:v4.8)
-FROM registry.ci.openshift.org/ocp/4.8:cli AS origincli
+FROM quay.io/openshift/origin-cli:4.8 AS origincli
```
Comment (vimalk78) [inline Dockerfile:37]: suggest to use 
```
registry.redhat.io/openshift4/ose-cli:v4.10 
```

File: Dockerfile [hunk 4]
```diff
@@ -29,14 +29,15 @@ COPY ${REMOTE_SOURCE}/apis ./apis
...
 
 #@follow_tag(registry.redhat.io/ubi8:latest)
-FROM registry.ci.openshift.org/ocp/4.8:base
+FROM registry.access.redhat.com/ubi8
```
Comment (vimalk78) [inline Dockerfile:40]: suggest to use
```
registry.redhat.io/ubi8:8.5 
```

File: origin-meta.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 from:
   - source: registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder\:v(?:[\.0-9\-]*).*
-    target: registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.8 AS builder
+    target: registry.access.redhat.com/ubi8/go-toolset AS builder
   - source: registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-cli:v4.8.0-([0-9]*).*
-    target: registry.ci.openshift.org/ocp/4.8:cli AS origincli
+    target: quay.io/openshift/origin-cli:4.8 AS origincli
   - source: registry.redhat.io/ubi8:8.(\d)-([\.0-9])*
-    target: registry.ci.openshift.org/ocp/4.8:base
+    target: registry.access.redhat.com/ubi8
```
Comment (vimalk78) [inline origin-meta.yaml:7]: same as above comments

File: Dockerfile (hunk 2 above)
Comment (jcantrill) [reply Dockerfile:32]: Build failed making a directory upon change.  I'm open to suggestions.  If this is only in the "builder" image then maybe it's a non-issue as I would expect the "run" base does not use this user.  Do you know?

File: Dockerfile (hunk 3 above)
Comment (jcantrill) [reply Dockerfile]: We know for certain these images are public?

Comment (jcantrill) [reply Dockerfile]: this one requires agreement to subscription: ```Before downloading or using this Container, you must agree to the Red Hat subscription agreement located at [redhat.com/licenses](https://www.redhat.com/en/about/agreements). If you do not agree with these terms, do not download or```

File: Dockerfile (hunk 4 above)
Comment (jcantrill) [reply Dockerfile]: this one is fine `Terms & conditions
The Red Hat Universal Base Image is free to deploy on Red Hat or non-Red Hat platforms and freely redistributable. S`

File: Dockerfile (hunk 3 above)
Comment (vimalk78) [reply Dockerfile:37]: how do you get this error?

File: Dockerfile.patch [hunk 6]
```diff
@@ -1,3 +0,0 @@
-22a23,24
-> RUN rm /etc/yum.repos.d/*
-> ADD art.repo /etc/yum.repos.d
```
Comment (vimalk78) [inline Dockerfile.patch:3]: we can remove `art.repo` also if it is not needed

File: Dockerfile (hunk 3 above)
Comment (jcantrill) [reply Dockerfile:37]: This is not an error but the verbiage displayed on the catalog UI

**Pull request merged**